### PR TITLE
Adds proxy argument for routing HttpLogger through a proxy

### LINKED ISSUE #205: Support zipkin server behind a proxy in HttpLogger
**Problem**
We have two networks (let A and B) that can only communicate with each other through a proxy gateway. When specifying the address to the Zipkin server in network B from an application in network A, the HttpLogger is attempting to connect directly without going through the proxy, resulting in a failure to post the trace.

Problem experienced using version `0.12.0` of the `zipkin-transport-http` package using Node v8.10.0 for Ubuntu 16.04.

**Solution**
I am working on implementing the solution outlined in the `node-fetch` repository under [node-fetch Options](https://github.com/bitinn/node-fetch#options) to be able to specify a `http.agent` in the fetch parameter.

### PR DESCRIPTION
**Pull request by zephinzer**
Fixes #205

### CODE REVIEW

File: packages/zipkin-transport-http/package.json [hunk 1]
```diff
@@ -13,12 +13,15 @@
   "license": "Apache-2.0",
   "repository": "https://github.com/openzipkin/zipkin-js",
   "dependencies": {
+    "http-proxy-agent": "^2.1.0",
```
Comment (adriancole) [inline packages/zipkin-transport-http/package.json]: what's the risk of having these dependencies?

**Comment by zephinzer** [comment]
sorry, it's my first time hearing that question, how should i assess that?

**Comment by adriancole** [comment]
On Mon, Apr 16, 2018 at 6:44 PM, Joseph Matthias Goh <
notifications@github.com> wrote:

> sorry, it's my first time hearing that question, how should i assess that?
>

So, I guess I'm assuming that this transport is used by a lot of users
users. Sometimes dependencies can conflict with application code,
especially if they change incompatibly a lot or pull a string of deps that
can also conflict. Sortof asking for some summary judgement

**Comment by zephinzer** [comment]
i see, the `http-proxy-agent` and `https-proxy-agent` together adds 7 dependencies including itself. the five dependencies are `debug`, `ms`, `agent-base`, `es6-promise` and `es6-promisify` which are widely used libraries, i don't assess it is likely there will be such conflicts caused by irregular release cycles/improper versioning. also, the proxy is only invoked upon passing of the argument so it will not break existing implementations

**Comment by adriancole** [comment]
do you anticipate this would break those using react or browser functionality? cc @DanielMSchmidt  @micaelsampaio

**Comment by zephinzer** [comment]
@adriancole i suspect it might, didn't consider that- are there any known open source projects using browser functionality so i can quickly replicate it to test?

**Comment by jcchavezs** [comment]
@zephinzer could you proceed with the test you mention?

**Comment by adriancole** [comment]
lacking an automatic  browser test, it would in my opinion suffice if you manually test by locally publishing this, editing the zipkin-js-example to use a proxy, and testing it

https://github.com/openzipkin/zipkin-js-example

I don't actually know of an automatic browserify test, and how to stitch it into circleci... maybe @eirslett does perhaps, but seems a separate issue to create that infra cc @openzipkin/devops-tooling

File: packages/zipkin-transport-http/package.json (hunk 1 above)
Comment (jcchavezs) [reply packages/zipkin-transport-http/package.json]: Do you mean security, amount of dependencies or compatibility?

**Comment by jcchavezs** [comment]
@zephinzer could you please rebase this? Also, any chance on continuing this?

**Comment by jcchavezs** [comment]
@zephinzer could we rebase this?

**Comment by jcchavezs** [comment]
@zephinzer could you please rebase this? Also, any chance on continuing this?

**Comment by jcchavezs** [comment]
Any chance instead we add the dependencies, we just add an interface for proxy and allow users to inject it?

**Comment by adriancole** [comment]
decoration somehow sounds good.

strategically: bear in mind we probably will want to redo the http logger internals soon, so that the "sender" concept is extracted. This will reduce the code about http to very small and makes proxy or alternate impls easy.

**Comment by jcchavezs** [comment]
This is fixed in https://github.com/openzipkin/zipkin-js/pull/439 hence I am closing this issue. Thanks @zephinzer anyways for the effort!

**Pull request closed without merging**